### PR TITLE
[Settings] Allow s/w decode for GBM platform

### PR DESCRIPTION
**Pull request by peak3d**
[Settings] Allow s/w decode for GBM platform

## Description
There are several situations in which forcing to the s/w (ffmpeg) path for video playback could make sense:

- issues in h/w decoder for a specific codec
- user wants to benefit from advanced options (tone-mapping / hq scaler ...)
- if users report decoding issues we usually ask if the issue apears with s/w too. Reason is that ffmpeg path is in general prooved.

This PR adds a setting entry in Settings::VideoPlayer::Process which allows deactivation of h/w decoder to force ffmpeg s/w path.

## Motivation and Context
Investigating decoder issues on odroid-N1 

## How Has This Been Tested?
Aarch64 gbm build on odroid-N1

## Types of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Improvement (non-breaking change which improves existing functionality)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

**Comment by peak3d** [comment]
This change affects both areas VP and GBM, therefore both of you as reviewers.

File: xbmc/utils/StringUtils.h [hunk 1]
```diff
@@ -46,6 +46,9 @@
 #include "XBDateTime.h"
 #include "utils/params_check_macros.h"
 
+#define STR_HELPER(x) #x
+#define STR(x) STR_HELPER(x)
```
Comment (Rechi) [inline xbmc/utils/StringUtils.h]: what's this for?

Comment (peak3d) [reply xbmc/utils/StringUtils.h]: for converting a compiler definition to string, see gbm.cmake

Comment (Rechi) [reply xbmc/utils/StringUtils.h]: I understand that, but why don't you add the defines in xbmc/settings/Settings.cpp?

Comment (peak3d) [reply xbmc/utils/StringUtils.h]: IMO these 2 macros are StringUtils, but I don't insist on that location.
Any place is fine for me.

Comment (MrMC) [reply xbmc/utils/StringUtils.h]: Should heavily comment this usage of pre-processor magic. And it should have a very descriptive name that reflects the specialized use case. "STR" is way too generic.

**Review by FernetMenta — APPROVED** [review]
+1, all our hw decoders can be turned off

File: xbmc/utils/StringUtils.h (hunk 1 above)
Comment (Rechi) [reply xbmc/utils/StringUtils.h]: If you expect that this marco will be needed at other places too, keep it here, but it definitely has to be commented.

**Comment by peak3d** [comment]
@lrusak any objections?

jenkins build this please

**Pull request merged**